app/services/fuzzySearchService.js

```diff
@@ -15,27 +17,24 @@ const fuseSearchOptions = {
 const getPaymentData = async ({ searchString, limit, offset, sortBy, filterBy, action }) => {
   if (!searchString) throw new Error('Empty search content')
 
-  const searchResults = await filterAndSearch(searchString, filterBy)
-  if (!searchResults.length) {
-    return { count: 0, rows: [] }
-  }
-
-  let results = getSortedResults(searchResults, sortBy)
-  if (action !== 'download') {
-    results = results.slice(offset, parseInt(offset) + parseInt(limit))
+  const searchResults = await search(searchString)
+  const filteredResults = applyFiltersAndGroupByPayee(searchResults, filterBy)
+  if (!filteredResults.length) {
+    return { count: 0, rows: [], filterOptions: {} }
   }
 
+  const sortedResults = getSortedResults(filteredResults, sortBy)
   return {
-    count: searchResults.length,
-    rows: removeFilterFields(results)
+    count: sortedResults.length,
+    rows: (action !== 'download') ? sortedResults.slice(offset, parseInt(offset) + parseInt(limit)) : sortedResults,
```

Comment:
From readability point of view, can we leave it as a separate line as it was earlier?

Reply:
I thought that was readable enough as we either slice the results or not depending on the action. But happy to move it back if it makes it easier to read.